soil_api/models/soil_property.py

```diff
@@ -0,0 +1,233 @@
+    bdod = 100
+    cec = 10
+    cfvo = 10
+    clay = 10
+    nitrogen = 100
+    ocd = 10
+    ocs = 10
+    phh2o = 10
+    sand = 10
+    silt = 10
+    soc = 10
+
+
+class SoilLayerList(BaseModel):
+    layers: List[SoilLayer] = Field(..., description="The queried soil property layers")
+
+
+class SoilPropertyUnit(BaseModel):
+    d_factor: SoilPropertiesConversionFactors = Field(
+        ..., description="The conversion factor", example=10
+    )
+    mapped_units: SoilPropertiesMappedUnits = Field(
+        ..., description="The mapped unit of the soil property", example="cm³/dm³"
+    )
+    target_units: SoilPropertiesTargetUnits = Field(
+        ..., description="The target unit of the soil property", example="m³/ha"
+    )
+    uncertainty_unit: str = Field(
+        ..., description="The unit of the uncertainty", example=""
+    )
+
+
+class SoilLayer(BaseModel):
+    code: SoilPropertiesCodes = Field(
+        ..., description="The soil property code", example="bdod"
+    )
+    name: SoilPropertiesNames = Field(
+        ..., description="The name of the soil property", example="Bulk density"
+    )
+    unit_measure: SoilPropertyUnit = Field(
+        ..., description="The unit of the soil property"
+    )
+    depths: List[SoilDepth] = Field(
+        ..., description="The queried soil depths with values"
+    )
+
+
+class SoilPropertyValueTypes(Enum):
+    mean = "mean"
+    Q0_05 = "Q0.05"
+    Q0_5 = "Q0.5"
+    Q0_95 = "Q0.95"
+    uncertainty = "uncertainty"
+
+
+class SoilPropertyValues(BaseModel):
+    mean: float | None = Field(
+        None, description="The mean value of the soil property", example=50
+    )
+    Q0_05: float | None = Field(
+        None,
+        description="The 5th percentile of the soil property",
+        example=40,
+        alias="Q0.05",
+    )
+    Q0_5: float | None = Field(
+        None,
+        description="The 50th percentile of the soil property",
+        example=50,
+        alias="Q0.5",
+    )
+    Q0_95: float | None = Field(
+        None,
+        description="The 95th percentile of the soil property",
+        example=60,
+        alias="Q0.95",
+    )
+    uncertainty: float | None = Field(
+        None, description="The uncertainty of the soil property", example=5
+    )
+
+
+class SoilDepthLabels(Enum):
+    depth_0_5 = "0-5cm"
+    depth_0_30 = "0-30cm"
+    depth_5_15 = "5-15cm"
+    depth_15_30 = "15-30cm"
+    depth_30_60 = "30-60cm"
+    depth_60_100 = "60-100cm"
+    depth_100_200 = "100-200cm"
+
+
+# Create a reverse lookup dictionary
+REVERSE_DEPTH_LOOKUP = {
+    v.value: k for k, v in list(SoilDepthLabels.__members__.items())
+}
+
+
+# Function to get the enum member using the reverse lookup
+def get_soil_depth_from_label(depth_string: str) -> SoilDepthLabels:
+    try:
+        return REVERSE_DEPTH_LOOKUP[depth_string]
+    except KeyError:
+        raise ValueError(f"No soil depth label found for '{depth_string}'")
+
+
+class SoilDepthTop(Enum):
```

Comment:
Could SoilDepthTop, SoilDepthBottom, and SoilDepthUnits just be a single dictonary with SoilDepthLabels as keys?

Using tuples
```python
soil_depth_dict = {
  SoilDepthLabels.depth_0_5: (0, 5, "cm")
  SoilDepthLabels.depth_0_30: (0, 30, "cm")
  ...
}

top_depth, bottom_depth, depth_unit = soil_depth_dict[SoilDepthLabels.depth_0_5]
```

If you want it to be more self-documenting you could also consider just having a dictonary instead of tuples for the values